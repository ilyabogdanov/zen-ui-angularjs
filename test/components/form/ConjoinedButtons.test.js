const div = document.createElement("div");
document.body.appendChild(div);

describe("directive: zen-conjoined-buttons", function() {
    let element, scope;
    beforeEach(angular.mock.module("ZenUI"));
    beforeEach(inject(function($compile, $rootScope) {
        scope = $rootScope.$new();
        element = ""+
            "<zen-conjoined-buttons>" +
            "<zen-link-button zen-href=\"::'___'\"></zen-link-button>" +
            "<zen-link-button zen-href=\"::'___'\"></zen-link-button>" +
            "<zen-link-button zen-href=\"::'___'\"></zen-link-button>" +
            "</zen-conjoined-buttons>";
        element = $compile(element)(scope);
        scope.$digest();
        angular.element(div).append(element);
    }));
    afterEach(function () {
        while (div.firstChild) {
            div.removeChild(div.firstChild);
        }
    });

    it("should render", function () {
        expect(div.childNodes).toHaveLength(1);
        expect(div.firstChild).toMatchSnapshot();
    });
});
