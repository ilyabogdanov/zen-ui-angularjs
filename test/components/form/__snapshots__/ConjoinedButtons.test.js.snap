// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directive: zen-conjoined-buttons should render 1`] = `
<div
  class="zen_ui__conjoined_buttons ng-scope"
  ng-transclude=""
>
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    zen-href="::'___'"
  >
    <a
      class="zen_ui__button"
      data-palette="ordinary"
      href="unsafe:___"
      ng-class="innerClazz"
      ng-href="___"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          />
        </span>
      </span>
    </a>
    <div />
  </div>
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    zen-href="::'___'"
  >
    <a
      class="zen_ui__button"
      data-palette="ordinary"
      href="unsafe:___"
      ng-class="innerClazz"
      ng-href="___"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          />
        </span>
      </span>
    </a>
    <div />
  </div>
  <div
    class="zen_ui__button_container ng-scope ng-isolate-scope"
    data-stretch="false"
    ng-class="outerClazz"
    zen-href="::'___'"
  >
    <a
      class="zen_ui__button"
      data-palette="ordinary"
      href="unsafe:___"
      ng-class="innerClazz"
      ng-href="___"
    >
      <span
        class="zen_ui__button_text"
      >
        <span>
          <span
            ng-transclude=""
          />
        </span>
      </span>
    </a>
    <div />
  </div>
</div>
`;
